crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -914,6 +913,128 @@ mod tests {
         assert!(candidates.get(&tiny_coin_id).is_none());
     }
 
+    #[tokio::test]
+    async fn test_insufficient_balance_will_retry_success() {
+        let test_cluster = TestClusterBuilder::new().build().await.unwrap();
+        let address = test_cluster.get_address_0();
+        let mut context = test_cluster.wallet;
+        let gases = get_current_gases(address, &mut context).await;
+        let config = FaucetConfig::default();
+
+        let reasonable_value = (config.num_coins as u64 * config.amount) * 10;
+        let _res = SuiClientCommands::SplitCoin {
+            coin_id: *gases[0].id(),
+            amounts: Some(vec![reasonable_value]),
+            gas_budget: 50000000,
+            gas: None,
+            count: None,
+            serialize_unsigned_transaction: false,
+            serialize_signed_transaction: false,
+        }
+        .execute(&mut context)
+        .await;
```

Comment:
Is the `_res` to avoid the complaint about an unused `Result`?  If so, it's better to `expect` or `unwrap` the result so we don't silently ignore the failure.

```suggestion
        SuiClientCommands::SplitCoin {
            coin_id: *gases[0].id(),
            amounts: Some(vec![reasonable_value]),
            gas_budget: 50000000,
            gas: None,
            count: None,
            serialize_unsigned_transaction: false,
            serialize_signed_transaction: false,
        }
        .execute(&mut context)
        .await
        .expect("Split failed!");
```